drivers/include/srf04.h

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) 2018 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_srf04 srf04 ultra sonic range finder
+ * @ingroup     drivers_sensors
+ * @brief       Device driver for the srf04 ultra sonic range finder
+ * @{
+ *
+ * @file
+ * @brief       Device driver for the srf04 ultra sonic range finder
+ *
+ * @author      Semjon Kerner <semjon.kerner@fu-berlin.de>
+ */
+
+#ifndef SRF04_H
+#define SRF04_H
+
+#include <stdint.h>
+#include <stdio.h>
+
+#include "periph/gpio.h"
+#include "xtimer.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief defines distance divisor
+ *
+ * @note for inch define distance as "1480"
+ */
+#ifndef SRF04_DISTANCE
+#define SRF04_DISTANCE  (584U)
+#endif
+
+/**
+ * @brief defines sensor required sample time
+ */
+#define SRF04_SAMPLE_PERIOD (50U * US_PER_MS)
+
+/**
+ * @brief   Status and error return codes
+ */
+enum {
+    SRF04_OK        =  0,   /**< exit without error */
+    SRF04_ERR_INVALID   = -1,   /**< error no valid measurement available*/
+    SRF04_ERR_MEASURING = -2,   /**< error sensor is measuring*/
+    SRF04_ERR_GPIO  = -3,   /**< error initializing gpio*/
+};
+
+/**
+ * @brief   GPIO pins for srf04 device
+ */
+typedef struct {
+    gpio_t trigger;
```

Comment:
The attributes of this struct are not documented, same for the `srf04_t` struct below.